azkaban-common/src/test/java/azkaban/executor/MockExecutorLoader.java

```diff
@@ -54,7 +54,8 @@
   @Override
   public void uploadExecutableFlow(ExecutableFlow flow)
       throws ExecutorManagerException {
-    flows.put(flow.getExecutionId(), flow);
+    ExecutableFlow exFlow = ExecutableFlow.createExecutableFlowFromObject(flow.toObject());
```

Comment:
> Why isn't it synchronized?

I guess it just doesn't matter too much. Job status is only changed by the designated JobRunner thread (at least in the cases that FlowRunnerTest covers). The FlowRunner thread keeps running until it sees that all jobs have reached a finished status before it ends the flow (and it will eventually see the changes in job statuses even though ExecutableNode.status is not synchronized).

Open questions maybe worth investigating:
- Does the FlowRunner set status for jobs that entirely crashed (exception in job thread)?
- Is it a problem that the flow is sometimes saved with some "outdated" job statuses?